added parameter default values

j-ed opened the pull request:
Fix #3069

Based on a source code analysis (looking for getSystemValue() function) I added the default values of the parameters.

- Unfortunately for several parameters different default values are used in the source code and my question how this should be addressed has not been answered yet. Examples:
```
getSystemValue('dbtype', 'sqlite')
getSystemValue('dbtype', '')
getSystemValue('dbtype')
```

- For other parameters contradictory values are used. Examples:
```
getSystemValue('preview_max_x', 2048)
getSystemValue('preview_max_x', 1024)
or
getSystemValue('preview_max_y', 2048)
getSystemValue('preview_max_y', 1024)
```

- Additionally some of the parameters have still not been adjusted for Nextcloud. Examples:
```
getSystemValue('dbname', 'owncloud')
getSystemValue('dbtableprefix', 'oc_')
```

I followed the recommendation of @nickvergessen and created separate issue tickets for each parameter.

j-ed committed:
added parameter default values

based on a source code analysis (looking for getSystemValue() function) I added the default values of the prameters

oparoz commented:
:+1:

MorrisJobke commented on config/config.sample.php:
```diff
@@ -74,8 +74,10 @@
 /**
- * Where user files are stored; this defaults to ``data/`` in the Nextcloud
- * directory. The SQLite database is also stored here, when you use SQLite.
+ * Where user files are stored.The SQLite database is also stored here, when
```
Space before `The` missing

j-ed replied:
[x] fixed.

j-ed committed:
Update config.sample.php

nickvergessen commented on config/config.sample.php:
````diff
@@ -762,6 +762,11 @@
  * seen in the first-run wizard and on Personal pages.
+ *
+ * Defaults to
+ * * Desktop client: ```https://nextcloud.com/install/#install-clients```
````
should be two ticks only on each side, not 3

j-ed committed:
replaced three backticks by two.

MorrisJobke commented:
@j-ed Sorry for not bringing this up earlier: we require a sign-off for your commits. You can read the details here: https://github.com/nextcloud/server/blob/master/CONTRIBUTING.md#sign-your-work

Thanks for the contribution and it now looks good. Once the sign-off is added to the commits we can merge it :)

MorrisJobke commented:
> Thanks for the contribution and it now looks good. Once the sign-off is added to the commits we can merge it :)

Then also CI gets green ;)

nickvergessen commented on config/config.sample.php:
```diff
@@ -1182,12 +1322,16 @@
+ *
+ * Defaults to ``2.2.0``
  */
 'minimum.supported.desktop.version' => '2.0.0',
```
I'd also adjust the sample value here to match the default.

j-ed replied:
fixed typo after double-checking which default is used in the source code.

nickvergessen commented on config/config.sample.php:
```diff
@@ -1153,17 +1287,23 @@
  * If you are applying a theme to Nextcloud, enter the name of the theme here.
  * The default location for themes is ``nextcloud/themes/``.
+ *
+ * Default to ``themes/`` in Nextcloud directory
```
No, this is not the theme directory, but the actual theme. This has no default value:
`Defaults to the theming app which is shipped since Nextcloud 9`

j-ed replied:
[x] done.

nickvergessen commented on config/config.sample.php:
```diff
@@ -1140,6 +1272,8 @@
  *
  * The Web server user must have write access to this directory.
+ *
+ * Defaults to ``''`` (empty string)
```
defaults to null

j-ed replied:
default information removed.

nickvergessen commented on config/config.sample.php:
```diff
@@ -1025,6 +1148,8 @@
  *
  * One way to test is applying for a trystack account at http://trystack.org/
+ *
+ * Defaults to ``''`` (empty string)
```
It's an array and has no default value.

j-ed replied:
[x] done.

ChristophWurst commented:
> nickvergessen requested changes

I've adjusted the label accordingly

j-ed committed:
applied various fixes

- removed default value for 'objectstore' because it has not default value.
- removed default value for 'tempdirectory' because it is unset.
- changed default value for 'theme' to "Defaults to the theming app which is shipped since Nextcloud 9"
- fixed typo in default value of 'minimum.supported.desktop.version', after double-checking the version
  number in /apps/dav/lib/Connector/Sabre/BlockLegacyClientPlugin.php:71.

Signed-off-by: Juergen Edner <juergen@eisfair.org>

j-ed commented:
@MorrisJobke I've added a 'Signed-off-by:" line to my last commit.

MorrisJobke commented:
> @MorrisJobke I've added a 'Signed-off-by:" line to my last commit.

It needs to be done for all commits. I will squash them into one and merge this here then.

MorrisJobke commented:
> It needs to be done for all commits. I will squash them into one and merge this here then.

Done in https://github.com/nextcloud/server/pull/3133
